File: Leetcode/stack_queue/1021_RemoveOutermostParentheses/Solution.py
class Solution(object):
    def removeOuterParentheses(self, S):
        """
        :type S: str
        :rtype: str
        """
        stk = []
        count, leftmost =0, 0
        for i in xrange(0, len(S)):
            count += 1 if S[i]=='(' else -1
            stk.append(S[i])
            if count == 0:
                stk.pop(leftmost)
                stk.pop()
                leftmost = len(stk)
        return "".join(stk)

File: Leetcode/stack_queue/1021_RemoveOutermostParentheses/Solution.java
class Solution {
    public String removeOuterParentheses(String S) {
        StringBuilder stack = new StringBuilder();
        int count = 0, leftmost = 0;
        for(Character c: S.toCharArray()){
            count += (c=='(')? 1: -1;
            stack.append(c);
            if(count==0){
                stack.deleteCharAt(leftmost);
                stack.setLength(stack.length() - 1);
                leftmost = stack.length();
            }
        }
        return stack.toString();
    }
}
